### PR TITLE
Rewrite docs for new builder process

### PR DESCRIPTION
**Pull request by AydinHassan**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -645,15 +630,16 @@ $subMenuBuilder = (new CliMenuBuilder)
...
 ```
 
-In this case `addSubMenu` will return the main menu builder, not the sub menu builder.
-
-The submenu menu item will be an instance of `\PhpSchool\CliMenu\MenuItem\MenuMenuItem`. If you need access to the submenu,
+Note: The submenu menu item will be an instance of `\PhpSchool\CliMenu\MenuItem\MenuMenuItem`. If you need access to the submenu,
 you can get it via `$menuMenuItem->getSubMenu()`.
 
+Note: The closure used to build the sub menu is also binded with the `CliMenuBuilder` instance so you can add items and such using `$this->addItem()` rather than using the function
+parameter, this method is slightly more concise, but if you are using an IDE it may be that code completion does not work when referring to `$this`.
```
Comment (Lynesth) [inline README.md]: Line break ?

File: README.md [hunk 2]
```diff
@@ -675,26 +662,25 @@ $itemCallable = function (CliMenu $menu) {
...
 $menu->open();
 ```
 
 There are a few things to note about the syntax and builder process here:
 
-1. There is no argument to `addSplitItem`.
-2. `addSplitItem` returns an instance of `SplitItemBuilder`, the main menu will take care of building all split items (and any submenu they might contain).
-3. You can call `addItem`, `addSubMenu` and `addStaticItem` on the `SplitItemBuilder`.
-4. Like when building submenus, you can call `end()` on the Split Item `SplitItemBuilder` instance to get the parent `CliMenuBuilder` back again. This is useful for chaining.
+1. The first parameter to `addSubMenu` is a closure, which will be invoked with a new instance of `SplitItemBuilder` which you can use to add items to the split item.
```
Comment (Lynesth) [inline README.md]: addSplitItem not addSubMenu

File: README.md [hunk 3]
```diff
@@ -675,26 +662,25 @@ $itemCallable = function (CliMenu $menu) {
...
+2. You can call `addItem`, `addSubMenu` and `addStaticItem` on the `SplitItemBuilder`. 
+3. `SplitItemBuilder` has a fluent interface so you can chain method calls.
```
Comment (Lynesth) [inline README.md]: Add a note that like the addSubMenu, the closure used to build the splititem is also binded with the `SplitItemBuilder` blablabla

File: README.md (hunk 1 above)
Comment (AydinHassan) [reply README.md]: There is one :)

**Pull request merged**